src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/ClientCreds/JsonWebToken.cs

```diff
@@ -202,6 +206,9 @@ internal class JWTPayload
 
             [DataMember(Name = JsonWebTokenConstants.ReservedClaims.JwtIdentifier, IsRequired=false, EmitDefaultValue=false)]
             public string JwtIdentifier { get; set; }
+
+			[DataMember(Name = JsonWebTokenConstants.ReservedClaims.Additional, IsRequired = false, EmitDefaultValue = false)]
+	        public Dictionary<string, string> AdditionalClaims { get; set; }
```

Comment:
how will the dictionary be serialized? also the question is how much work is required on the service side to support the JWT with the new "add" claim.

Reply:
see unit test on the serialization question. yeah, the server side and public api changes both needs to be discussed further

Reply:
Consider IDictionary instead of Dictionary.
Have you set tabs as spaces?